maro/rl/policy_v2/policy_base.py

```diff
@@ -212,22 +213,23 @@ def get_batch_loss(self, batch: dict, explicit_grad: bool = False) -> object:  #
         """
         pass
 
-    @abstractmethod
     def data_parallel(self, *args, **kwargs) -> None:
         """"Initialize a proxy in the policy, for data-parallel training.
         Using the same arguments as `Proxy`."""
-        pass
+        self.task_queue_client = TaskQueueClient()
+        self.task_queue_client.create_proxy(*args, **kwargs)
 
     def data_parallel_with_existing_proxy(self, proxy: Proxy) -> None:
         """"Initialize a proxy in the policy with an existing one, for data-parallel training."""
-        self._proxy = proxy
+        self.task_queue_client = TaskQueueClient()
+        self.task_queue_client.set_proxy(proxy)
 
     def exit_data_parallel(self) -> None:
-        if self._proxy is not None:
-            self._proxy.close()
+        if hasattr(self, "task_queue_client"):
```

Comment:
Could we avoid using `hasattr`? In my personal opinion, the better way is create a private attribute `_task_queue_client` with type `Optional[TaskQueueClient]`, and call it if it is not `None`.